Disable GI probe capturing lights with bake mode disabled

aqnuep opened the pull request:
The bake mode property of lights previously didn't affect GI probes.
This change makes the GI probe ignore lights that have their bake mode
set to disabled.

aqnuep commented:
Disabling GI probe capture for certain dynamic lights is useful to avoid flickering artifacts.

One question may be whether we should add a separate property to control this or using the existing bake mode property, as proposed by this change, is fine.

reduz commented on drivers/gles3/rasterizer_storage_gles3.h:
```diff
@@ -966,6 +968,7 @@ class RasterizerStorageGLES3 : public RasterizerStorage {
 	virtual VS::LightType light_get_type(RID p_light) const;
 	virtual float light_get_param(RID p_light, VS::LightParam p_param);
 	virtual Color light_get_color(RID p_light);
+	virtual bool light_get_bake_gi(RID p_light);
```
Technically this is not baking but using, so I would change the function names from
light_xxx_bake_gi
to
light_xx_use_gi

aqnuep replied:
Well, technically it is baked, just not only once, but I get what you mean :)

Otherwise you're fine with the bake mode property of the node itself affecting this, or you'd like to add a new property for the light that explicitly controls GI use?

Calinou replied:
I think using the same property for GIProbe and BakedLightmap is fine, as you're unlikely to use both at the same time anyway.

aqnuep replied:
Done.

reduz reviewed (dismissed):
Other than that, the PR is great

aqnuep committed:
Disable GI probe capturing lights with bake mode disabled

The bake mode property of lights previously didn't affect GI probes.
This change makes the GI probe ignore lights that have their bake mode
set to disabled.

akien-mga commented:
Thanks!
